src/config/from_document.rs

```diff
@@ -329,3 +336,89 @@ impl TryFrom<ServiceDocument> for Config {
     from_document(value)
   }
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
You can instead write tests using graphql files. Checkout the integration `/tests` folder.

Reply:
Done. Realized that there are some new changes in integration tests. After fixing them, it is all fine and covers this feature. Help review, thanks!